temporian/core/operators/drop_index.py

```diff
@@ -0,0 +1,220 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""DropIndex operator."""
+from typing import List, Optional, Union
+
+from temporian.core import operator_lib
+from temporian.core.data.event import Event
+from temporian.core.data.feature import Feature
+from temporian.core.data.sampling import Sampling
+from temporian.core.operators.base import Operator
+from temporian.proto import core_pb2 as pb
+
+
+class DropIndexOperator(Operator):
+    def __init__(
+        self,
+        event: Event,
+        index_names: List[str],
+        keep,
+    ) -> None:
+        super().__init__()
+
+        # input event
+        self.add_input("event", event)
+
+        # attributes
+        self.add_attribute("index_names", index_names)
+        self.add_attribute("keep", keep)
+
+        # output features
+        output_features = self._generate_output_features(
+            event, index_names, keep
+        )
+
+        # output sampling
+        output_sampling = Sampling(
+            index=[
+                index_name
+                for index_name in event.sampling().index()
+                if index_name not in index_names
+            ]
+        )
+        # output event
+        self.add_output(
+            "event",
+            Event(
+                features=output_features,
+                sampling=output_sampling,
+                creator=self,
+            ),
+        )
+        self.check()
+
+    def _generate_output_features(
+        self, event: Event, index_names: List[str], keep: bool
+    ) -> List[Feature]:
+        output_features = [
+            Feature(name=feature.name(), dtype=feature.dtype())
+            for feature in event.features()
+        ]
+        if keep:
+            for feature_name in index_names:
+                # check no other feature exists with this name
+                if feature_name in event.feature_names:
+                    raise ValueError(
+                        f"Feature name {feature_name} coming from index already"
+                        " exists in event."
+                    )  # TODO: add automatic suffix instead of raising error? add capability to rename index
+
+                output_features.append(
+                    Feature(name=feature_name, dtype=None)  # TODO: fix dtype
+                )
+
+        return output_features
+
+    @property
+    def dst_feat_names(self) -> List[str]:
+        feature_names = self.inputs()["event"].feature_names
+        return (
+            self.attributes()["index_names"] + feature_names
+            if self.attributes()["keep"]
+            else feature_names
+        )
+
+    @property
+    def dst_index_names(self) -> List[str]:
+        return [
+            name
+            for name in self.inputs()["event"].sampling().index()
+            if name not in self.attributes()["index_names"]
```

Comment:
I would cache the dictionary query i.e. stored "self.attributes()["index_names"]" in a variable.